geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/StatSerializer.scala

```diff
@@ -273,15 +281,16 @@ object KryoStatSerializer {
     }
   }
 
-  private [stats] def readZ3Histogram(input: Input, sft: SimpleFeatureType, immutable: Boolean): Z3Histogram = {
+  private [stats] def readZ3Histogram(input: Input, sft: SimpleFeatureType, immutable: Boolean, version: Int): Z3Histogram = {
     val geomIndex = input.readInt(true)
     val dtgIndex  = input.readInt(true)
+    val period = if (version > 1) TimePeriod.withName(input.readString()) else TimePeriod.Week
     val length = input.readInt(true)
 
     val stat = if (immutable) {
-      new Z3Histogram(geomIndex, dtgIndex, length) with ImmutableStat
+      new Z3Histogram(geomIndex, dtgIndex, period, length) with ImmutableStat
     } else {
-      new Z3Histogram(geomIndex, dtgIndex, length)
+      new Z3Histogram(geomIndex, dtgIndex, period, length)
```

Comment:
Ok, just for my sanity, did the various v1 stats have a time bound implied?  And now we are just being explicit about it?